test/test_datasets.py

```diff
@@ -490,6 +480,35 @@ def inject_fake_data(self, tmpdir, config):
         return num_images_per_category * len(categories)
 
 
+class ImageNetTestCase(datasets_utils.ImageDatasetTestCase):
+    DATASET_CLASS = datasets.ImageNet
+    REQUIRED_PACKAGES = ['scipy']
+    CONFIGS = datasets_utils.combinations_grid(split=('train', 'val'))
+
+    def inject_fake_data(self, tmpdir, config):
+        tmpdir = pathlib.Path(tmpdir)
+
+        wnid = 'n01234567'
+        if config['split'] == 'train':
+            datasets_utils.create_image_folder(
+                root=tmpdir,
+                name=tmpdir / 'train' / wnid / wnid,
+                file_name_fn=lambda image_idx: f"{wnid}_{image_idx}.JPEG",
+                num_examples=1,
```

Comment:
Nit: Instead of hard coding it here, maybe set `num_examples` within each branch and simply return this. Bonus: If you change the number of examples dependent on the split you get a little better "coverage" with little cost.